responsibleai/responsibleai/feature_metadata.py

```diff
@@ -1,57 +1,66 @@
 # Copyright (c) Microsoft Corporation
 # Licensed under the MIT License.
 
-import warnings
 from typing import Any, Dict, List, Optional
 
 from responsibleai.exceptions import UserConfigValidationException
 
 
 class FeatureMetadata:
-    def __init__(self, identity_feature_name: Optional[str] = None,
-                 datetime_features: Optional[List[str]] = None,
+    def __init__(self,
+                 identity_feature_name: Optional[str] = None,
+                 time_column_name: Optional[str] = None,
                  categorical_features: Optional[List[str]] = None,
-                 dropped_features: Optional[List[str]] = None):
+                 dropped_features: Optional[List[str]] = None,
+                 time_series_id_column_names: Optional[List[str]] = None):
         """Placeholder class for feature metadata provided by the user.
 
         :param identity_feature_name: Name of the feature which helps to
                                       uniquely identify a row or instance
                                       in user input dataset.
         :type identity_feature_name: Optional[str]
-        :param datetime_features: List of datetime features in the
-                                  user input dataset.
-        :type datetime_features: Optional[List[str]]
+        :param time_column_name: name of datetime feature in the user input
```

Comment:
Isn't datetime more common in terms of usage for classification and regression tasks? 

The time_column is very specific to forecasting. I would suggest retaining datetime_features and adding time_column_name instead.

Also we should stick to the existing nomenclature here of using *_feature_name/*_feature_names. Maybe you can use time_feature_name.

Reply:
There may be models that can take datetime columns, but it's not at all clear how we would handle that in the UI as we assume everything is numbers only. All string columns get converted to numbers before getting passed into the UI. This time index column is handled separately for time series which is why that doesn't cause issues, but it's not at all clear how this would work for other tasks. It's also very much out of scope for my workstream. I will make a note to raise this as a discussion point with the broader team shortly. For now, I can make it an optional list, but if anyone specifies more than 1 feature or it's not forecasting then I'll raise an exception during validation saying this isn't supported yet. Wdyt?

I was going for compatible names with AutoML which uses the ones I have in the PR. If we care about doing it differently I can certainly change it. I doubt it'll confuse anyone too much.

Reply:
Made the changes I mentioned above.

Reply:
Resolving since I made the changes and started the discussion internally.